sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/GaugeResult.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.metrics;
+
+import com.google.auto.value.AutoValue;
+
+/**
+ * The result of a {@link Gauge} metric.
+ */
+@AutoValue
+public abstract class GaugeResult {
+  public abstract long value();
+
+  public static final GaugeResult ZERO = create(-1L);
```

Comment:
Why is -1 the ZERO? Isn't it valid for the reported value to be -1?

Could we instead include a timestamp on the result (the approximate time the value was taken), which would allow the ZERO to have a nonsensical timestamp?